### PR TITLE
[libsndfile] Update to 1.2.2

### PR DESCRIPTION
**Pull request by evpobr**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [ ] The "supports" clause reflects platforms that may be fixed by this new version
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ ] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by xiaozhuai** [comment]
This port has install `Findxxx.cmake` for its dependencies, which should be avoided. We need patch it to follow the usage provided by vcpkg.
See https://github.com/libsndfile/libsndfile/blob/2c6ffea6ac8bb1b89e1f778bdad8bf8b0342d32f/CMakeLists.txt#L853-L856

**Comment by FrankXie05** [comment]
https://github.com/libsndfile/libsndfile/blob/2c6ffea6ac8bb1b89e1f778bdad8bf8b0342d32f/CMakeLists.txt#L821

**Comment by dg0yt** [comment]
> vcpkg conflicts with upstream regarding dependency `mp3lame`.

No, upstream is broken, it lacks keyword `TARGET` at the failing line:
https://github.com/libsndfile/libsndfile/blob/2c6ffea6ac8bb1b89e1f778bdad8bf8b0342d32f/cmake/Findmp3lame.cmake#L61

**Comment by evpobr** [comment]
@FrankXie05, ready.

**Comment by xiaozhuai** [comment]
@evpobr We need remove all the `Findxxx.cmake` in portfile.cmake before `vcpkg_cmake_configure`

**Comment by evpobr** [comment]
@FrankXie05, ready.

**Review by FrankXie05 — DISMISSED** [review]

**Comment by dg0yt** [comment]
`libsndfile[*]:x64-linux` doesn't build. Please test all features and CMake usage (with a symbol from the lib!) in x64-linux which is a static triplet. 

I didn't look at the (implementation) details but I think the brief comments about avoiding the find modules were misleading. Available CMake configs usually aren't drop-in replacements. They have different names, provide targets instead of  variables, and using the targets during the build implies using `find_dependency(...)` in the exported config. It at least unlikely that non of the removed Find modules needs more action in the patch.

**Comment by evpobr** [comment]
@dg0yt , what do you mean?

https://github.com/microsoft/vcpkg/pull/34388/checks?check_run_id=17729086236

**Comment by dg0yt** [comment]
It builds in CI, with default features. This is necessary, but is it sufficient?

- Feature `regtest` failed before in 1.2.0, so this isn't a regression. Let's look at `libsndfile[experimental]` which is everything minus `regtest`. `experimental` maps to using/supporting speex.
- libsndfile development is somewhat coupled to vcpkg. Both projects share contributors, and the libsndfile build system has cmake code which reacts to vcpkg variables.
- The portfile passes `CMAKE_FIND_PACKAGE_PREFER_CONFIG=ON`, and that's why the find modules were not used when a config for a package name was available.

This explains why it is not as bad as I initially thought. It already uses the config and targets, and it has find_dependency. There are still some rough edges.

This is from the 1.2.0 config log:
~~~
-- Could NOT find ALSA (missing: ALSA_LIBRARY ALSA_INCLUDE_DIR) 
...
-- Found Speex: /vcpkg/installed/x64-linux/debug/lib/libspeex.a (found version "1.2.1") 
~~~
- Not finding ALSA is a (installation order) problem and a port bug, given port `alsa`, but that's another story.
- Speex (for `experimental`) is found.

The new version should do the same:
~~~
-- Could NOT find ALSA (missing: ALSA_LIBRARY ALSA_INCLUDE_DIR) 
...
CMake Warning at /vcpkg/scripts/buildsystems/vcpkg.cmake:859 (_find_package):
  By not providing "FindSpeex.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Speex", but
  CMake did not find one.

  Could not find a package configuration file provided by "Speex" with any of
  the following names:

    SpeexConfig.cmake
    speex-config.cmake

  Add the installation prefix of "Speex" to CMAKE_PREFIX_PATH or set
  "Speex_DIR" to a directory containing one of the above files.  If "Speex"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  cmake/SndFileChecks.cmake:60 (find_package)
  CMakeLists.txt:95 (include)
~~~
- Not finding ALSA is not a regression, so we don't care now.
- It no longer finds Speex. *This is a regression from this PR.* It isn't reported a build error because it is not marked as required.
  The fastest fix for the regression is to not remove this Find module. It isn't one which would get installed.
  (The best solution is to move it to config mode, similar to the other packages. But this task is more involved.)

File: ports/libsndfile/portfile.cmake [hunk 1]
```diff
@@ -1,11 +1,22 @@
...
 
+file(REMOVE "${SOURCE_PATH}/cmake/FindFLAC.cmake")
+file(REMOVE "${SOURCE_PATH}/cmake/Findmp3lame.cmake")
+file(REMOVE "${SOURCE_PATH}/cmake/FindMpg123.cmake")
+file(REMOVE "${SOURCE_PATH}/cmake/FindOgg.cmake")
+file(REMOVE "${SOURCE_PATH}/cmake/FindOpus.cmake")
+file(REMOVE "${SOURCE_PATH}/cmake/FindSndio.cmake")
+file(REMOVE "${SOURCE_PATH}/cmake/FindSpeex.cmake")
+file(REMOVE "${SOURCE_PATH}/cmake/FindVorbis.cmake")
+
```
Comment (dg0yt) [inline ports/libsndfile/portfile.cmake:19]:
```suggestion
```

File: ports/libsndfile/001-avoid-installing-find-modules.patch [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ 	endif ()
+ 
+ 	if(ENABLE_MPEG)
+ 		set (SndFile_WITH_MPEG 1)
+-		list (APPEND FIND_MODULES_INSTALL_LIST 
+-			${CMAKE_CURRENT_LIST_DIR}/cmake/Findmpg123.cmake
+-			${CMAKE_CURRENT_LIST_DIR}/cmake/Findmp3lame.cmake)
+ 	else ()
+ 		set (SndFile_WITH_MPEG 0)
+ 	endif ()
```
Comment (dg0yt) [inline ports/libsndfile/001-avoid-installing-find-modules.patch:25]:
```suggestion
```

**Review by dg0yt — COMMENTED** [review]
AFAICS it is enough to not install the find modules.

File: ports/libsndfile/001-avoid-installing-find-modules.patch (hunk 2 above)
Comment (evpobr) [reply ports/libsndfile/001-avoid-installing-find-modules.patch:25]: It's better to leave it as it is. I use `SndFile_WITH_EXTERNAL_LIBS` and `SndFile_WITH_MPEG` in `SndFileConfig.cmake`.

**Comment by evpobr** [comment]
@dg0yt , did i miss something ot it is ok?

**Comment by dg0yt** [comment]
You missed the point of my comment regarding the patch.
Remove the first chunk of the patch. That chunk removes an effect on `FIND_MODULES_INSTALL_LIST` which already has no effect due the last chunk of the patch.
And the variables you want are in the code with or without the patch.

**Review by dg0yt — APPROVED** [review]
LGTM with regard to the update.
(The port has issues but IMO this out of scope for this PR.)

**Review by FrankXie05 — APPROVED** [review]

**Pull request merged**